Hmm this seems suspicious in the log:
```
Found results file /Users/dotnet-bot/dotnetbuild/work/4bc49cfb-4e54-4e80-b956-4085f63855a8/Work/3794520e-9043-4e3c-bf3d-120e76f5d6ef/Exec/TestResults/_dci-mac-build-080_2019-04-30_01_55_20_332.trx with format trx
Uploading results in batches of size 1000
Traceback (most recent call last):
  File "/Users/dotnet-bot/dotnetbuild/work/4bc49cfb-4e54-4e80-b956-4085f63855a8/Payload/reporter/run.py", line 95, in <module>
    main()
  File "/Users/dotnet-bot/dotnetbuild/work/4bc49cfb-4e54-4e80-b956-4085f63855a8/Payload/reporter/run.py", line 84, in main
    for b in batches:
  File "/Users/dotnet-bot/dotnetbuild/work/4bc49cfb-4e54-4e80-b956-4085f63855a8/Payload/reporter/helpers.py", line 13, in batch
    for item in iterable:
  File "/Users/dotnet-bot/dotnetbuild/work/4bc49cfb-4e54-4e80-b956-4085f63855a8/Payload/reporter/test_results_reader/__init__.py", line 94, in <genexpr>
    return (add_logs(tr, log_list) for tr in f.read_results(file_path))
  File "/Users/dotnet-bot/dotnetbuild/work/4bc49cfb-4e54-4e80-b956-4085f63855a8/Payload/reporter/formats/trx.py", line 41, in read_results
    classname = test_classes[test_name]
KeyError: 'Microsoft.AspNetCore.Mvc.ApplicationModels.AttributeRouteModelTests.ReplaceTokens_InvalidFormat(template: "text]", values: [], reason: "Token delimiters (\'[\', \']\') are imbalanced.")'
+ date -u +%FT%TZ
2019-04-30T08:55:31Z
+ /usr/local/bin/python2 /Users/dotnet-bot/dotnetbuild/work/4bc49cfb-4e54-4e80-b956-4085f63855a8/Payload/xunit-reporter.py
2019-04-30 01:55:32,174: ERROR: xunit-reporter(69): main: Unable to report xunit results: no test results xml file found.
```

Maybe the uploading of the results went awry.

--- a/.azure/pipelines/helix-test.yml
+++ b/.azure/pipelines/helix-test.yml
@@ -17,6 +17,8 @@ jobs:
     steps:
     - script: .\build.cmd -all -ci /t:Helix /p:IsHelixJob=true /bl:$env:BUILD_SOURCESDIRECTORY\artifacts\logs\SendToHelix.binlog
       displayName: Run build.cmd helix target
+      env:
+        SYSTEM_ACCESSTOKEN: $(System.AccessToken) # We need to set this env var to publish helix results to Azure Dev Ops
     artifacts:
     - name: Helix_logs
       path: artifacts/logs/

--- a/build/repo.targets
+++ b/build/repo.targets
@@ -151,7 +151,7 @@
              Properties="$(BuildProperties);__DummyTarget=GenerateBuildAssetManifest" />
   </Target>
 
-  <Target Name="Helix" DependsOnTargets="Restore">
+  <Target Name="Helix">
     <MSBuild Projects="$(MSBuildThisFileDirectory)..\eng\helix\helix.proj"
              Targets="Test"
              Properties="$(BuildProperties);$(SolutionProperties);"

--- a/eng/Version.Details.xml
+++ b/eng/Version.Details.xml
@@ -388,9 +388,9 @@
       <Uri>https://github.com/dotnet/arcade</Uri>
       <Sha>ea9b2530a1559dd6426730aba52fca1924db2b65</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.DotNet.Helix.Sdk" Version="2.0.0-beta.19251.6">
+    <Dependency Name="Microsoft.DotNet.Helix.Sdk" Version="2.0.0-beta.19253.2">
       <Uri>https://github.com/dotnet/arcade</Uri>
-      <Sha>ea9b2530a1559dd6426730aba52fca1924db2b65</Sha>
+      <Sha>e31dd0f6fd12a136a2310a308d235c360703b221</Sha>
     </Dependency>
     <Dependency Name="Microsoft.AspNetCore.Testing" Version="3.0.0-preview6.19252.2" CoherentParentDependency="Microsoft.EntityFrameworkCore">
       <Uri>https://github.com/aspnet/Extensions</Uri>

--- a/eng/helix/helix.proj
+++ b/eng/helix/helix.proj
@@ -15,7 +15,7 @@
     <HelixBuild Condition=" '$(CI)' == 'true' ">$(BUILD_BUILDNUMBER)</HelixBuild>
     <WaitForWorkItemCompletion Condition=" '$(CI)' == 'true' ">false</WaitForWorkItemCompletion>
     <FailOnMissionControlTestFailure>false</FailOnMissionControlTestFailure>
-    <EnableAzurePipelinesReporter>false</EnableAzurePipelinesReporter>
+    <EnableAzurePipelinesReporter>true</EnableAzurePipelinesReporter>
     <IsExternal>true</IsExternal>
     <Creator>aspnetcore</Creator>
     <SkipInvalidConfigurations>true</SkipInvalidConfigurations>

--- a/eng/helix/vstest/runtests.cmd
+++ b/eng/helix/vstest/runtests.cmd
@@ -36,7 +36,7 @@ REM only puts the explicit filter traits the user provided in
 REM Filter syntax: https://github.com/Microsoft/vstest-docs/blob/master/docs/filter.md
 set NONFLAKY_FILTER="Flaky:All!=true&Flaky:Helix:All!=true&Flaky:Helix:Queue:All!=true&Flaky:Helix:Queue:%HELIX%!=true"
 echo Running non-flaky tests.
-%DOTNET_ROOT%\dotnet vstest %target% --logger:trx --logger:console;verbosity=normal --TestCaseFilter:%NONFLAKY_FILTER%
+%DOTNET_ROOT%\dotnet vstest %target% --logger:trx --TestCaseFilter:%NONFLAKY_FILTER%
 if errorlevel 1 (
     echo Failure in non-flaky test 1>&2
     set exit_code=1
@@ -45,7 +45,7 @@ if errorlevel 1 (
 
 set FLAKY_FILTER="Flaky:All=true|Flaky:Helix:All=true|Flaky:Helix:Queue:All=true|Flaky:Helix:Queue:%HELIX%=true"
 echo Running known-flaky tests.
-%DOTNET_ROOT%\dotnet vstest %target% --logger:trx --logger:console;verbosity=normal --TestCaseFilter:%FLAKY_FILTER%
+%DOTNET_ROOT%\dotnet vstest %target% --logger:trx --TestCaseFilter:%FLAKY_FILTER%
 if errorlevel 1 (
     echo Failure in flaky test 1>&2
     REM DO NOT EXIT and DO NOT SET EXIT_CODE to 1

--- a/eng/helix/vstest/runtests.sh
+++ b/eng/helix/vstest/runtests.sh
@@ -79,7 +79,6 @@ if [ $nonflaky_exitcode != 0 ]; then
     echo "Non-flaky tests failed!" 1>&2
     # DO NOT EXIT
 fi
-
 FLAKY_FILTER="Flaky:All=true|Flaky:Helix:All=true|Flaky:Helix:Queue:All=true|Flaky:Helix:Queue:$HELIX=true"
 echo "Running known-flaky tests."
 $DOTNET_ROOT/dotnet vstest $1 --logger:trx --TestCaseFilter:"$FLAKY_FILTER"

--- a/eng/targets/Helix.targets
+++ b/eng/targets/Helix.targets
@@ -52,7 +52,7 @@
 
   <Target Name="_CreateHelixWorkItem" Condition="$(BuildHelixPayload)">
 
-    <MSBuild Projects="$(MSBuildProjectFullPath)" Targets="_PublishHelixArchive" RemoveProperties="HelixTargetQueue" />
+    <MSBuild Projects="$(MSBuildProjectFullPath)" Targets="_PublishHelixArchive" RemoveProperties="HelixTargetQueue;TestRunId" />
 
     <ConvertToAbsolutePath Paths="$(PublishDir)">
       <Output TaskParameter="AbsolutePaths" PropertyName="PublishAbsoluteDir" />

--- a/global.json
+++ b/global.json
@@ -4,6 +4,6 @@
   },
   "msbuild-sdks": {
     "Yarn.MSBuild": "1.13.0",
-    "Microsoft.DotNet.Helix.Sdk": "2.0.0-beta.19251.6"
+    "Microsoft.DotNet.Helix.Sdk": "2.0.0-beta.19253.2"
   }
 }

--- a/src/Servers/Kestrel/Kestrel/test/GeneratedCodeTests.cs
+++ b/src/Servers/Kestrel/Kestrel/test/GeneratedCodeTests.cs
@@ -17,12 +17,12 @@ namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
         [Flaky("https://github.com/aspnet/AspNetCore-Internal/issues/2223", FlakyOn.Helix.All)]
         public void GeneratedCodeIsUpToDate()
         {
-            var repositoryRoot = typeof(GeneratedCodeTests).Assembly.GetCustomAttributes<AssemblyMetadataAttribute>().First(f => string.Equals(f.Key, "RepositoryRoot", StringComparison.OrdinalIgnoreCase)).Value;
+            var repositoryRoot = TestPathUtilities.GetSolutionRootDirectory("Kestrel");
 
-            var httpHeadersGeneratedPath = Path.Combine(repositoryRoot, "src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs");
-            var httpProtocolGeneratedPath = Path.Combine(repositoryRoot, "src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.Generated.cs");
-            var httpUtilitiesGeneratedPath = Path.Combine(repositoryRoot, "src/Servers/Kestrel/Core/src/Internal/Infrastructure/HttpUtilities.Generated.cs");
-            var transportConnectionGeneratedPath = Path.Combine(repositoryRoot, "src/Servers/Kestrel/Transport.Abstractions/src/Internal/TransportConnection.Generated.cs");
+            var httpHeadersGeneratedPath = Path.Combine(repositoryRoot, "Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs");
+            var httpProtocolGeneratedPath = Path.Combine(repositoryRoot, "Kestrel/Core/src/Internal/Http/HttpProtocol.Generated.cs");
+            var httpUtilitiesGeneratedPath = Path.Combine(repositoryRoot, "Kestrel/Core/src/Internal/Infrastructure/HttpUtilities.Generated.cs");
+            var transportConnectionGeneratedPath = Path.Combine(repositoryRoot, "Kestrel/Transport.Abstractions/src/Internal/TransportConnection.Generated.cs");
 
             var testHttpHeadersGeneratedPath = Path.GetTempFileName();
             var testHttpProtocolGeneratedPath = Path.GetTempFileName();

--- a/src/Servers/Kestrel/Kestrel/test/GeneratedCodeTests.cs
+++ b/src/Servers/Kestrel/Kestrel/test/GeneratedCodeTests.cs
@@ -17,12 +17,12 @@ namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
         [Flaky("https://github.com/aspnet/AspNetCore-Internal/issues/2223", FlakyOn.Helix.All)]
         public void GeneratedCodeIsUpToDate()
         {
-            var repositoryRoot = TestPathUtilities.GetSolutionRootDirectory("Kestrel");
+            var repositoryRoot = typeof(GeneratedCodeTests).Assembly.GetCustomAttributes<AssemblyMetadataAttribute>().First(f => string.Equals(f.Key, "RepositoryRoot", StringComparison.OrdinalIgnoreCase)).Value;
 
-            var httpHeadersGeneratedPath = Path.Combine(repositoryRoot, "Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs");
-            var httpProtocolGeneratedPath = Path.Combine(repositoryRoot, "Kestrel/Core/src/Internal/Http/HttpProtocol.Generated.cs");
-            var httpUtilitiesGeneratedPath = Path.Combine(repositoryRoot, "Kestrel/Core/src/Internal/Infrastructure/HttpUtilities.Generated.cs");
-            var transportConnectionGeneratedPath = Path.Combine(repositoryRoot, "Kestrel/Transport.Abstractions/src/Internal/TransportConnection.Generated.cs");
+            var httpHeadersGeneratedPath = Path.Combine(repositoryRoot, "src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs");
+            var httpProtocolGeneratedPath = Path.Combine(repositoryRoot, "src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.Generated.cs");
+            var httpUtilitiesGeneratedPath = Path.Combine(repositoryRoot, "src/Servers/Kestrel/Core/src/Internal/Infrastructure/HttpUtilities.Generated.cs");
+            var transportConnectionGeneratedPath = Path.Combine(repositoryRoot, "src/Servers/Kestrel/Transport.Abstractions/src/Internal/TransportConnection.Generated.cs");
 
             var testHttpHeadersGeneratedPath = Path.GetTempFileName();
             var testHttpProtocolGeneratedPath = Path.GetTempFileName();

--- a/src/Servers/Kestrel/Kestrel/test/Microsoft.AspNetCore.Server.Kestrel.Tests.csproj
+++ b/src/Servers/Kestrel/Kestrel/test/Microsoft.AspNetCore.Server.Kestrel.Tests.csproj
@@ -1,4 +1,4 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <TargetFramework>netcoreapp3.0</TargetFramework>
@@ -9,11 +9,6 @@
     <Compile Include="$(KestrelSharedSourceRoot)test\*.cs" LinkBase="shared" />
     <Compile Include="$(SharedSourceRoot)test\SkipOnHelixAttribute.cs" />
     <Content Include="$(KestrelSharedSourceRoot)test\TestCertificates\*.pfx" LinkBase="shared\TestCertificates" CopyToOutputDirectory="PreserveNewest" />
-
-    <AssemblyAttribute Include="System.Reflection.AssemblyMetadataAttribute">
-      <_Parameter1>RepositoryRoot</_Parameter1>
-      <_Parameter2>$(RepositoryRoot)</_Parameter2>
-    </AssemblyAttribute>
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETFramework'">

--- a/src/Servers/Kestrel/Kestrel/test/Microsoft.AspNetCore.Server.Kestrel.Tests.csproj
+++ b/src/Servers/Kestrel/Kestrel/test/Microsoft.AspNetCore.Server.Kestrel.Tests.csproj
@@ -9,6 +9,11 @@
     <Compile Include="$(KestrelSharedSourceRoot)test\*.cs" LinkBase="shared" />
     <Compile Include="$(SharedSourceRoot)test\SkipOnHelixAttribute.cs" />
     <Content Include="$(KestrelSharedSourceRoot)test\TestCertificates\*.pfx" LinkBase="shared\TestCertificates" CopyToOutputDirectory="PreserveNewest" />
+
+    <AssemblyAttribute Include="System.Reflection.AssemblyMetadataAttribute">
+      <_Parameter1>RepositoryRoot</_Parameter1>
+      <_Parameter2>$(RepositoryRoot)</_Parameter2>
+    </AssemblyAttribute>
   </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFrameworkIdentifier)' != '.NETFramework'">

--- a/src/Servers/Kestrel/test/Interop.FunctionalTests/H2SpecCommands.cs
+++ b/src/Servers/Kestrel/test/Interop.FunctionalTests/H2SpecCommands.cs
@@ -16,6 +16,33 @@ namespace Interop.FunctionalTests
 {
     public static class H2SpecCommands
     {
+        #region chmod
+        // user permissions
+        const int S_IRUSR = 0x100;
+        const int S_IWUSR = 0x80;
+        const int S_IXUSR = 0x40;
+
+        // group permission
+        const int S_IRGRP = 0x20;
+        const int S_IWGRP = 0x10;
+        const int S_IXGRP = 0x8;
+
+        // other permissions
+        const int S_IROTH = 0x4;
+        const int S_IWOTH = 0x2;
+        const int S_IXOTH = 0x1;
+
+        const int _0755 =
+            S_IRUSR | S_IXUSR | S_IWUSR
+            | S_IRGRP | S_IXGRP
+            | S_IROTH | S_IXOTH;
+
+        [DllImport("libc", SetLastError = true)]
+        private static extern int chmod(string pathname, int mode);
+
+        private static int chmod755(string pathname) => chmod(pathname, _0755);
+        #endregion
+
         private const int TimeoutSeconds = 15;
 
         private static string GetToolLocation()
@@ -27,11 +54,15 @@ namespace Interop.FunctionalTests
             }
             else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
             {
-                return Path.Combine(root, "linux", "h2spec");
+                var toolPath = Path.Combine(root, "linux", "h2spec");
+                chmod755(toolPath);
+                return toolPath;
             }
             else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
             {
-                return Path.Combine(root, "darwin", "h2spec");
+                var toolPath = Path.Combine(root, "darwin", "h2spec");
+                chmod755(toolPath);
+                return toolPath;
             }
             throw new NotImplementedException("Invalid OS");
         }

